### PR TITLE
Thermals V3

### PR DESCRIPTION
**Pull request by AdamTadeusz**
## Description

-------> SHADERS AND NEW MATERIALS IN https://github.com/NeotokyoRebuild/neoAssets/pull/15 <-------

Added noise to the thermals;

Reduced initial thermal shader colour gradient offset to make dark objects stand out a bit more (using black screen with default colour of 0 0 0 to work out what I believe is the correct value) and added a multiplier to compensate for bright areas not being as bright with the offset reduction;

View models and other models now use the tvmgrad and tvmgrad2 textures respectively to colour correct. This methodology can probably be used in the night vision and especially the motion vision to get a more original colour and also make those shaders more performant and less complicated (see materials/dev/nvgrad.vtf and materials/dev/mvgrad.vtf);

The way the gradients are passed from the material to the shader can also probably be used to remove the janky way that thermoptic cloak's strength is passed to the thermoptic shader, instead of through the current modification of a convar everytime a cloaked object is drawn

## Toolchain

- Windows MSVC VS2022
- Linux GCC Distro Native [Specify distro + GCC version]
- Linux GCC 10 Sniper 3.0

### CODE REVIEW

File: mp/src/materialsystem/stdshaders/neo_thermalvision_ps30.fxc [hunk 1]
```diff
@@ -16,6 +19,7 @@ struct PS_INPUT
 float4 main( const PS_INPUT i ) : COLOR
 {    
 	const float4 fbColor = tex2D(FrameBuffer, i.texCoord);
-    const float avgBright = (fbColor.r + fbColor.g + fbColor.b) * 0.3333;    
-    return tex2D(TvTexture, float2(min(1.0, avgBright + g_XOffset), 0.5)) * g_BrightnessScale;
+	const float noise =  0.9 + (tex2D(NoiseTexture, i.texCoord + float2(g_XNoiseOffset, g_YNoiseOffset)) / 10);
```
Comment (brysondev) [inline mp/src/materialsystem/stdshaders/neo_thermalvision_ps30.fxc:22]: Nit tabs

File: mp/src/materialsystem/stdshaders/neo_thermalvision_ps30.fxc [hunk 2]
```diff
@@ -16,6 +19,7 @@ struct PS_INPUT
 float4 main( const PS_INPUT i ) : COLOR
 {    
 	const float4 fbColor = tex2D(FrameBuffer, i.texCoord);
```
Comment (brysondev) [inline mp/src/materialsystem/stdshaders/neo_thermalvision_ps30.fxc:21]: Nit tabs

**Review by brysondev — DISMISSED** [review]

**Review by Rainyan — CHANGES_REQUESTED** [review]

File: mp/src/materialsystem/stdshaders/Neo_ThermalVision.cpp [hunk 3]
```diff
@@ -16,7 +17,8 @@ BEGIN_SHADER_PARAMS
 SHADER_PARAM(FBTEXTURE, SHADER_PARAM_TYPE_TEXTURE, "_rt_ThermalVision", "")
 //SHADER_PARAM(BLURTEXTURE, SHADER_PARAM_TYPE_TEXTURE, "_rt_SmallHDR0", "")
 SHADER_PARAM(TVTEXTURE, SHADER_PARAM_TYPE_TEXTURE, "dev/tvgrad2", "")
-//SHADER_PARAM(NOISETEXTURE, SHADER_PARAM_TYPE_TEXTURE, "dev/noise", "")
+SHADER_PARAM(NOISETEXTURE, SHADER_PARAM_TYPE_TEXTURE, "dev/noise", "")
+SHADER_PARAM(NOISETRANSFORM, SHADER_PARAM_TYPE_VEC3, "Vector(0, 0, 0)", "")
```
[the remark this answers is not in the document]
Comment (AdamTadeusz) [reply mp/src/materialsystem/stdshaders/Neo_ThermalVision.cpp:21]: This tutorial https://developer.valvesoftware.com/wiki/Source_SDK_2013:_Your_Second_Shader uses brackets, I've also seen just a few numbers with no brackets somewhere else in the wiki

**Review by brysondev — APPROVED** [review]

**Review by Rainyan — APPROVED** [review]
:+1:

**Pull request merged**